sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -274,7 +284,7 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
   override def visitInsertOverwriteTable(
       ctx: InsertOverwriteTableContext): InsertTableParams = withOrigin(ctx) {
     assert(ctx.OVERWRITE() != null)
-    val tableIdent = visitTableIdentifier(ctx.tableIdentifier)
+    val tableIdent = visitMultipartIdentifier(ctx.multipartIdentifier)
```

Comment:
I think this needs to be updated to remove the `ParseException` thrown when `IF NOT EXISTS` is present and there are dynamic partitions. I think that is an analysis problem, not a parse problem.

Also, I don't see a reason why IF NOT EXISTS would not be supported with dynamic partitions. Wouldn't that fail if any partitions would be overwritten? It seems to make sense to me, but maybe there is a good reason why this is not allowed? @gatorsmile can you comment?

Reply:
We discussed this in the DSv2 sync last night and decided to add a method to the write builder to pass this `IF NOT EXISTS` flag. This will be done in a follow-up to avoid over-complicating this commit.

Reply:
Opened https://issues.apache.org/jira/browse/SPARK-28374